### ISSUE TITLE
v9 interpolation bug when sharing a SpringValue between components

### ISSUE THREAD

**Issue by aleclarson** [react-spring/react-spring#745]
I think beta 30 totally breaks what I was doing with interpolate. Any reason to deprecate it? I have a few cases where I share the same value across a few different components/props to sync the animations, so a lot of lines like:

```
class Example extends React.Component {
  getBadgePositionInterpolation(controller: any, yMultiplier?: number = 1) {
    return controller.animated.spring
      .interpolate([0, 1], [20, 0])
      .interpolate(value => `translate(${value} ${value * yMultiplier})`);
  }

  getLowerBadgeStyles() {
    const {spring} = this.state.lowerBadgeMask.animated;
    return {
      opacity: spring.interpolate([0, 0.5, 1], [0, 0, 1]),
      transform: spring.interpolate(value => `translate(${16 - value * 16}px, ${16 - value * 16}px)`),
    };
  }

  getUpperBadgeStyles() {
    const {spring} = this.state.upperBadgeMask.animated;
    return {
      opacity: spring.interpolate([0, 0.5, 1], [0, 0, 1]),
      transform: spring.interpolate(value => `translate(${16 - value * 16}px, ${-1 * (16 - value * 16)}px)`),
    };
  }
}
```

So I am 99% sure this is a React Spring bug (I was on beta 20 before going to beta 30, so this could've been introduced anytime before then, can try bisecting betas more to see if I can find which version).

Anyways, for some context - I am doing some animations using a combination of SVG masks and dom elements.

I am animating an svg mask on the same spring value as a dom element's transform, to create this effect in beta20 and prior:

[good example](http://wtf.amadeus.wtf/monosnap/screencast_2019-07-03_15-27-16.mp4)

And after updating to beta30, you can see what is now happening:

[bug example](http://wtf.amadeus.wtf/monosnap/screencast_2019-07-03_15-21-28.mp4)

It appears the SVG is animating fine on the value (as can be seen by the mask) however the dom element just get's the `from` values applied and never animates.

_Originally posted by @amadeus in https://github.com/react-spring/react-spring/issues/642#issuecomment-508275776_

**Comment by aleclarson** [comment]
@amadeus This won't be fixed until we have a simplified repro.

**Comment by amadeus** [comment]
@aleclarson sorry for being slow to respond! I have finally found the bug and created a repro. It's actually not at all related to sharing a controller. It actually appears to be a race condition with effects and mounting elements on the same tick:

https://codesandbox.io/s/modest-brook-khgh8?fontsize=14

**Comment by aleclarson** [comment]
Fixed in `9.0.0-canary.809.3.a66b280`
